File: basic/1260.cpp
//1260

#include <cstdio>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

vector<int> graph[1001];
bool check[1001];

void dfs(int x)
{
    check[x] = true;
    printf("%d ", x);
    for (int i = 0; i < graph[x].size(); i++)
    {
        int node = graph[x][i];
        if (!check[node])
        {
            dfs(node);
        }
    }
}
void bfs(int x)
{
    queue<int> q;
    q.push(x);
    check[x] = true;
    printf("%d ", x);
    while (!q.empty())
    {
        int now = q.front();
        q.pop();
        for (int i = 0; i < graph[now].size(); i++)
        {
            int next = graph[now][i];
            if (!check[next])
            {
                check[next] = true;
                printf("%d ", next);
                q.push(next);
            }
        }
    }
}
int main()
{
    int n, m, v;
    scanf("%d %d %d", &n, &m, &v);
    for (int i = 0; i < m; i++)
    {
        int u, v;
        scanf("%d %d", &u, &v);
        graph[u].push_back(v);
        graph[v].push_back(u);
    }
    for (int i = 0; i <= n; i++)
    {
        sort(graph[i].begin(), graph[i].end());
    }
    dfs(v);
    printf("\n");
    fill(check, check + sizeof(check), 0);
    bfs(v);
}

File: Silver/1260.py

import sys


def input(): return sys.stdin.readline().rstrip()


def bfs(v):
    global check, graph
    check[v] = 1
    print(v, end=' ')
    for e in graph[v]:
        if check[e] == 0:
            check[e] = 1
            bfs(e)


def dfs(v):
    global check, graph
    check[v] = 1
    print(v, end=' ')
    q = []
    q.append(v)
    while len(q) != 0:
        now = q.pop(0)
        for e in graph[now]:
            if check[e] == 0:
                check[e] = 1
                print(e, end=' ')
                q.append(e)


(N, M, V) = map(int, input().split())
graph = [[] for _ in range(N+1)]

for i in range(M):
    (u, v) = map(int, input().split())
    graph[u].append(v)
    graph[v].append(u)

for i in range(1, N+1):
    graph[i].sort()

check = [0 for _ in range(N+1)]
bfs(V)
print()
check = [0 for _ in range(N+1)]
dfs(V)
